standard/variables.md

```diff
@@ -988,3 +998,225 @@ variable_reference
 ## 9.6 Atomicity of variable references
 
 Reads and writes of the following data types shall be atomic: `bool`, `char`, `byte`, `sbyte`, `short`, `ushort`, `uint`, `int`, `float`, and reference types. In addition, reads and writes of enum types with an underlying type in the previous list shall also be atomic. Reads and writes of other types, including `long`, `ulong`, `double`, and `decimal`, as well as user-defined types, need not be atomic. Aside from the library functions designed for that purpose, there is no guarantee of atomic read-modify-write, such as in the case of increment or decrement.
+
+## §ref-span-safety Reference variables and returns
+
+### §ref-span-safety-general General
```

Comment:
I'm having a hard time reviewing this due to the change of terminology.  Then I realized all I had to do was... https://gist.github.com/gafter/d6068f35deb2a558b08b478f9dba1b1f (part 2...)

I'm sorry I ran out of time, but I think it would be a good idea to go through that gist and check off the rules one by one when we can find the corresponding rule in this PR.

Reply:
Yes, that's a good exercise. I'm confident all the rules are included. When I added this section, I started with this section, and then applied the following edits:

1. Re-order local and parameter rules.
2. Change title of "method invocation" to "function invocation" because the same rules would apply to local function declarations.
3. "Rvalue" changes to "Value" for consistency with the standard's existing terms.
4. "Span constructor" and "default" rules move to structs.md, in the section of "safe context" for `ref struct` types. 

I did a quick scan, and none of the rules were removed, or had other semantic edits.

Reply:
Leaving this open to find an owner for going through the gist.